### DISCUSSION TITLE
Operation not permitted

### DISCUSSION THREAD

**Discussion by JoostdeJonge123** [zwave-js/zwave-js#5902] [Q&A]
Dear all,

I'm using the node-zwave-js already for a long time without any problems, perfect with domoticz. But since a few problems I get into the following problem, for some reason the /dev/ttyACM0 port does not open any more, giving the following error:

_2023-06-11T12:33:48.675Z DRIVER   starting driver...
2023-06-11T12:33:48.675Z DRIVER   opening serial port /dev/ttyACM0
2023-06-11T12:33:57.679Z DRIVER   Failed to open the serial port: Error: Operation not permitted, cannot open /dev/ttyACM0
2023-06-11T12:33:57.679Z DRIVER   destroying driver instance...
2023-06-11T12:33:57.680Z DRIVER   driver instance destroyed_

It seems like permissions to /dev/ttyACM0 are correct. I really do not know where to start looking. Any help is really appreciated.

Thanks.

Regards, Joost

**Comment by JoostdeJonge123** [comment]
Small update... using the zwave usb stick in different VM, everything is working... very strange...

**Reply by AlCalzone** [reply]
`Operation not permitted, cannot open ...` is an error that comes from the operating system. Searching for that gives some hints to fix this which may depend on your environment.

One general recommendation: prefer `/dev/serial/by-id/...` paths over `/dev/tty...` paths if possible.
